source/agora/utils/Test.d

```diff
@@ -627,60 +632,360 @@ unittest
+        The value drawn (`amount * toward.length`) must be lesser or equal
+        to the fund currently attached and in the refund transaction.
 
-    // Add support for Transactions with multiple recipients
-    foreach (idx, const ref _; input.outputs)
-        result.inputs ~= Input(inputHash, cast(uint) idx);
+        Params:
+            amount = `Amount` to distribute to each `PublicKey`
+            toward = Array of `PublicKey` to give `amount` to
 
-    scope sign = (KeyPair kp, Hash h) @trusted { return kp.secret.sign(h[]); };
+        Returns:
+            Reference to `this` for easy chaining
 
-    const resultHash = result.hashFull();
-    foreach (idx, ref in_; result.inputs)
+    ***************************************************************************/
+
+    public ref typeof(this) draw (Amount amount, in PublicKey[] toward... )
+        @safe return
+    {
+        assert(toward.length > 0, "No beneficiary in `draw` transaction");
+        assert(amount > Amount(0), "Cannot have outputs of value `0`");
+
+        foreach (key; toward)
+        {
+            this.data.outputs ~= Output(amount, key);
+            // Provide friendlier error message for developers
+            if (!this.leftover.value.sub(amount))
+                assert(0, format("Error: Withdrawing %d times %s BOA underflown",
+                                 toward.length, amount));
+        }
+
+        return this;
+    }
+
+    /***************************************************************************
+
+        Similar to `draw(Amount, PublicKey[])`, but uses all available funds
+
+        Note that if the available funds are not a multiple of `toward.length`,
+        the refund `Output` will holds the leftovers.
+
+        Params:
+            toward = Beneficiary to split the currently available amount between
+
+        Returns:
+            Reference to `this` for easy chaining
+
+    ***************************************************************************/
+
+    public ref typeof(this) split (scope const PublicKey[] toward...)
+        @safe return
     {
-        auto rng = from.find!(a => a.address == input.outputs[idx].address);
-        assert(rng.length);
-        const owner = rng[0];
-        in_.signature = sign(owner, resultHash);
+        assert(toward.length > 0, "No beneficiary in `split` transaction");
+
+        auto forEach = this.leftover.value;
+        const remainder = forEach.div(toward.length);
+        return this.draw(forEach, toward);
     }
-    return result;
+
+    /// Refund output for the transaction
+    private Output leftover;
+
+    /// Stores the inputs to consume until `sign` is called
+    private const(OutputRef)[] inputs;
+
+    /// Transactions to be built and returned
+    private Transaction data;
+}
+
+/// Returns: A range ofTransactions which are spendable in the Genesis block
+public auto genesisSpendable () @safe
+{
+    return GenesisBlock.txs.filter!(tx => tx.type == TxType.Payment);
+}
```

Comment:
Since in the future we want genesis to have both `Freezing` and `Payment` transactions, this is needed